lib/components/base/MarkdownEditor.vue

```diff
@@ -0,0 +1,698 @@
+      <Button icon-only :action="() => openLinkModal()" :disabled="previewMode || disabled">
+        <LinkIcon />
+      </Button>
+      <Button icon-only :action="() => openImageModal()" :disabled="previewMode || disabled">
+        <ImageIcon />
+      </Button>
+      <Button icon-only :action="() => openVideoModal()" :disabled="previewMode || disabled">
+        <YouTubeIcon />
+      </Button>
+      <Checkbox v-model="previewMode" description="Preview" class="preview">Preview</Checkbox>
+    </div>
+    <textarea
+      id="project-description"
+      ref="editorInput"
+      v-model="currentValue"
+      :disabled="disabled"
+      :class="{ hide: previewMode }"
+      @input="() => onInput()"
+      @keydown="(event) => onKeyDown(event)"
+    />
+    <div v-if="!previewMode" class="info-blurb">
+      <InfoIcon />
+      <span>
+        This editor supports
+        <a href="https://docs.modrinth.com/docs/markdown" target="_blank">Markdown formatting</a>.
+      </span>
+    </div>
+    <div
+      v-if="previewMode"
+      style="width: 100%"
+      class="markdown-body"
+      v-html="renderHighlightedString(currentValue ?? '')"
+    />
+  </div>
+</template>
+
+<script setup>
+import { computed, ref } from 'vue'
+import { renderHighlightedString } from '@/helpers/highlight.js'
+import Checkbox from '@/components/base/Checkbox.vue'
+import Modal from '@/components/base/Modal.vue'
+
+const emit = defineEmits(['update:modelValue'])
+
+const props = defineProps({
+  modelValue: {
+    type: String,
+    default: '',
+  },
+  disabled: {
+    type: Boolean,
+    default: false,
+  },
+  headingButtons: {
+    type: Boolean,
+    default: true,
+  },
+})
+
+const currentValue = ref(props.modelValue)
+const previewMode = ref(false)
+const editorInput = ref(null)
+
+const linkText = ref('')
+const linkUrl = ref('')
+
+function cleanUrl(url) {
+  if (!url.startsWith('https://')) {
+    url = 'https://' + url
+  }
+  return url
+}
+
+const linkMarkdown = computed(() => {
+  if (!linkUrl.value || !linkUrl.value.includes('.')) {
+    return ''
+  }
+  const url = cleanUrl(linkUrl.value)
+  return url ? `[${linkText.value ? linkText.value : url}](${url})` : ''
+})
+
+const imageAlt = ref('')
+const imageUrl = ref('')
+
+const imageMarkdown = computed(() => {
+  const url = cleanUrl(imageUrl.value)
+  return url ? `![${imageAlt.value}](${url})` : ''
+})
+
+const videoUrl = ref('')
+const youtubeRegex =
+  /^(?:https?:)?(?:\/\/)?(?:youtu\.be\/|(?:www\.|m\.)?youtube\.com\/(?:watch|v|embed)(?:\.php)?(?:\?.*v=|\/))([a-zA-Z0-9_-]{7,15})(?:[?&][a-zA-Z0-9_-]+=[a-zA-Z0-9_-]+)*$/
+
+const videoMarkdown = computed(() => {
+  const match = youtubeRegex.exec(videoUrl.value)
+  if (match) {
+    return `<iframe width="560" height="315" src="https://www.youtube.com/embed/${match[1]}" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>`
```

Comment:
Probably better to have it in nocookie mode [like so](https://www.youtube-nocookie.com/embed/d1YBv2mWll0)

```suggestion
    return `<iframe width="560" height="315" src="https://www.youtube-nocookie.com/embed/${match[1]}" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>`
```